### ISSUE TITLE
Option to hide shaped and shapeless recipes like regular crafting recipes

### ISSUE THREAD

**Issue by Guinaro** [codetaylor/artisan-worktables#102]
I would like the option to be able to hide recipes from JEI, just like I can do with regular crafting recipes.

.sethidden(true (default))

Reason:
I'm recreating the banner recipes functionality in a Tailor Workshop Table. But I only need demonstration recipes visible. Thus I want to show a banner pattern or banner symbol result only for white banners with black dye. All other recipes need to be hidden. For my shapeless recipes I can currently suffice with a single entry with <ore:dye> and get the meta in my Recipe function.
However I need to add all recipes for all dyes, to be able to do shaped patterns, as I cannot do a <ore:dye>, as I usually have many dyes necessary for a single pattern, and they need to be identical.
e.g. : Top Stripe Pattern in Strings.
```
"ts": [["dye",	"dye",	"dye",	"dye",	"dye"],
	["null",	"null",	"null",	"null",	"null"],
	["null",	"null",	"banner",	"null",	"null"],
	["null",	"null",	"null",	"null",	"null"],
	["null",	"null",	"null",	"null",	"null"]
], //Top Stripe
```
Currently for each dye color the recipe appears, and I want to limit it to JEI only showing white banner with black dye.

Guinaro

**Comment by codetaylor** [comment]
Feature will be available in 1.12.2-1.19.0  :)

**Comment by Guinaro** [comment]
Thank you for this very quick implementation.

**Comment by codetaylor** [comment]
Feature is available in 1.12.2-1.19.0-SNAPSHOT.1

**Comment by codetaylor** [comment]
I'm going to go ahead and close this early, even though it hasn't been merged into the master branch or properly released yet. It is available in the current alpha snapshot and will be in the next beta release.
